bundler/helpers/lib/functions/parent_dependency_resolver.rb

```diff
@@ -0,0 +1,55 @@
+# frozen_string_literal: true
+
+module Functions
+  class ParentDependencyResolver
+    def initialize(dependency_name:, target_version:, lockfile_name:)
+      @dependency_name = dependency_name
+      @target_version = target_version
+      @lockfile_name = lockfile_name
+
+      Bundler.settings.set_command_option("only_update_to_newer_versions", true)
```

Comment:
Concern: this seems to modify static state from a constructor.
It seems ~safe given the usage, but heuristically I would never expect a constructor to modify static state.

Ideas:
* shift to the top of `blocking_parent_dependencies`
* refactor to a static method